src/steps/endpoints.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  createDirectRelationship,
+  createIntegrationEntity,
+  createMappedRelationship,
+  Entity,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  Relationship,
+  RelationshipClass,
+  RelationshipDirection,
+} from '@jupiterone/integration-sdk-core';
+
+import { createAPIClient } from '../client';
+import {
+  CalculatedIntegrationConfig,
+  MalwareBytesEndpointWithAgent,
+} from '../types';
+import {
+  relationships,
+  entities,
+  ACCOUNT_ENTITY_DATA_KEY,
+  MAPPED_HOST_AGENT_PROTECTS_ENDPOINT,
+} from '../constants';
+
+export function getEndpointKey(id: string): string {
+  return `malwarebytes_endpoint:${id}`;
+}
+
+export function getHostAgentKey(id: string): string {
+  return `malwarebytes_host_agent:${id}`;
+}
+
+/**
+ * Creates a mapped relationship between a HostAgent and Host. This should not be
+ * used when the target is known to be an EC2 Host.
+ *
+ * @param serviceEntity the HostAgent that protects the host
+ * @param host a HostAsset that is protected by the HostAgent
+ */
+export function createHostAgentProtectsHostRelationship(
+  agentEntity: Entity,
+  host: MalwareBytesEndpointWithAgent,
+): Relationship {
+  return createMappedRelationship({
+    _class: RelationshipClass.PROTECTS,
+    _type: MAPPED_HOST_AGENT_PROTECTS_ENDPOINT,
+    _mapping: {
+      sourceEntityKey: agentEntity._key,
+      relationshipDirection: RelationshipDirection.FORWARD,
+      targetFilterKeys: [['_class', 'hostname']],
+      targetEntity: {
+        _class: 'Host',
+        _type: 'user_endpoint',
+        _key: getEndpointKey(host.machine.id),
```

Comment:
This looks OK to me. I know there was some discussion in the Google doc about whether or not to include a `_key` property here. As long as it's not present in the `targetFilterKeys` array, it's really up to you. I personally might have left it off, but that's nothing but my own preference.

Reply:
It's removed - we trust your and @aiwilliams preferences :+1: